### PR TITLE
chore: update to Electron 22

### LINKED ISSUE #144791: Voiceover when expanding selection could be better
Couple issues:
1. when expanding the ***** selection, it only ever says 1star star star` selected even when more *'s are shown in the text. This seems to be a macos global bug
2. The formatting `foo unselected, foo bar selected` is not great, it hides the most important info. This seems to be a chromium thing, repros in edge, but notes handles it better IMO

Recording 2022-03-09 at 14 14 46

Recording 2022-03-09 at 14 20 03

### LINKED ISSUE #154666: Linux: running with `sandbox` enabled out of sources fails to start after first start
This one puzzles me and I need help from you and/or maybe Alex to figure it out. Steps:
* setup Linux running out of sources to run with sandbox enabled (use internal builds)
* run once and enable sandbox mode
* after restart verify sandbox mode is enabled (e.g. try to type `process` in devtools)
* quit
* start again

I now see tons of errors printed:

```
[7771:0710/055522.227642:INFO:CONSOLE(142)] "Uncaught SyntaxError: Unexpected end of input", source: vscode-file://vscode-app/home/parallels/GitDevelopment/monaco/out/vs/workbench/contrib/mergeEditor/browser/view/colors.js (142)
```

image

A couple of thoughts:
* is it a possible cached data corruption (though it reproduces with cached data disabled)
* is it a problem with custom protocol handler and sandbox mode enabled?
* is it some other cache Chromium produces after a first startup?

### LINKED ISSUE #156723: Drag and drop does not work in Wayland
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: 1.69.2
- OS Version: Manjaro Linux, Kernel: 5.15, Wayland, WM: sway

Steps to Reproduce:

1. Add `--enable-features=UseOzonePlatform --ozone-platform=wayland` to code-flags.conf
2. Open code
3. Attempt to drag tabs, files, or anything else
4. Does not give any indication that they are being dragged, and the elements to not get moved around

I am unable to re-order windows, create multiple panes with drag and drop, etc

It works if I do not add the flags. The issue is I need the flags for scaling in Wayland.

### LINKED ISSUE #157184: The GUI is malfunctioned on macOS of VMware
Does this issue occur when all extensions are disabled?: Yes

- VS Code Version: from 1.60
- OS Version: macOS 12.4

Steps to Reproduce:

1.Install VMware workstation 16.2.3
2.Install macOS 12.4
3.Install VS code 1.50 which is normal
4.Install VS code 1.60, then the window is black and the texture rendering is bad. The GUI is not functioning properly.
All platform in Virtual Machine including Virtual Box are working properly. However, since 1.60, the GUI is malfunctioned on macOS with same condition on VMware. That's too severe to influence my work productivity. Please provide a solution and review this problem.
VScode

### LINKED ISSUE #163736: vscode menu cannot popup after maximize/restore window
Type: <b>Bug</b>

1. Open Help menu.
2. Double click the vscode window title to maximize window.
3. Try to open Help menu again.

VS Code version: Code 1.72.2 (d045a5eda657f4d7b676dedbfa7aab8207f8a075, 2022-10-12T22:16:26.920Z)
OS version: Linux x64 5.10.0-18-amd64
Modes:
Sandboxed: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD Ryzen 5 4500U with Radeon Graphics (6 x 3251)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: disabled_off<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>video_encode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled<br>webgpu: disabled_off|
|Load (avg)|1, 1, 0|
|Memory (System)|22.78GB (18.35GB free)|
|Process Argv|--unity-launch --crash-reporter-id 3baa2306-59ad-467a-b353-f56c4e72dfec|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|gnome|
|XDG_CURRENT_DESKTOP|GNOME|
|XDG_SESSION_DESKTOP|gnome|
|XDG_SESSION_TYPE|wayland|
</details><details><summary>Extensions (12)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-eslint|dba|2.2.6
svg-gallery|dev|2.1.0
xml|Dot|2.5.1
gitlens|eam|12.2.2
EditorConfig|Edi|0.16.4
prettier-vscode|esb|9.9.0
vscode-react-typescript|inf|1.3.1
svg|joc|1.4.20
vscode-docker|ms-|1.22.1
csharp|ms-|1.25.0
color-highlight|nau|2.5.0
bash-beautify|sha|0.1.1

</details>

### PR DESCRIPTION
**Pull request by deepak1556**
Refs https://github.com/microsoft/vscode/issues/166265
Fixes https://github.com/microsoft/vscode/issues/156723
Fixes https://github.com/microsoft/vscode/issues/163736
Fixes https://github.com/microsoft/vscode/issues/157184
Fixes https://github.com/microsoft/vscode/issues/154666
Fixes https://github.com/microsoft/vscode/issues/144791
Refs https://github.com/microsoft/vscode-encrypt/issues/33

### CODE REVIEW

**Comment by deepak1556** [comment]
This is blocked on Electron missing https://source.chromium.org/chromium/chromium/src/+/main:buildtools/third_party/libc++/trunk/include/__memory_resource/; headers from the libcxx bundle shipped.

**Comment by samuelstroschein** [comment]
@deepak1556 is this PR on track for the March release of VSCode? 

Helps us with planning the release of inlang's vs code extension, see https://github.com/inlang/inlang#ide-extension-get-notified. 

OFF-TOPIC I saw that you are a member of libgit2. Amusing coincidence, we are planning to switch to libgit2 for inlang https://github.com/inlang/inlang/issues/278.

File: build/lib/electron.ts
Comment (deepak1556) [inline build/lib/electron.ts:94]: I will update this to an official version once the PGO related builds are available in a follow-up PR. For now, lets merge with the custom version to get insiders coverage.

**Comment by deepak1556** [comment]
I will update distro once this is merged.

File: .yarnrc
Comment (bpasero) [inline .yarnrc:2]: By now `22.3.1` is out.

File: build/lib/electron.ts
Comment (bpasero) [inline build/lib/electron.ts:215]: Do we need to unwind this?

Comment (deepak1556) [reply build/lib/electron.ts:215]: Yup once builds are available, for now it is safe to merge with this version number.

File: .yarnrc
Comment (deepak1556) [reply .yarnrc]: Updated in  https://github.com/microsoft/vscode/pull/166430/commits/6175b0441a6336ebbf6b8c9cfde6d6eb97868811

**Review by bpasero — APPROVED** [review]

**Review by aeschli — APPROVED** [review]

**Comment by samuelstroschein** [comment]
Has this PR been reverted?

**Comment by deepak1556** [comment]
@samuelstroschein yes in favor of https://github.com/microsoft/vscode/pull/175843, we will post more context in the following week.

**Pull request merged**
